src/DssSpell.t.sol

```diff
@@ -495,5 +498,68 @@ contract DssSpellTest is DssSpellTestBase {
         // Validate post-spell state
         assertEq(arbitrumGateway.validDomains(arbDstDomain), 0, "l2-arbitrum-invalid-dst-domain");
     }
-}
 
+    function testFlapperUniV2() public {
+        // Create surplus manipulating dai and sin (ONLY FOR GOERLI)
+        stdstore.target(address(vat)).sig("dai(address)").with_key(address(vow)).depth(0).checked_write(60_000_000 * RAD);
+        stdstore.target(address(vat)).sig("sin(address)").with_key(address(vow)).depth(0).checked_write(5_000_000 * RAD);
+        stdstore.target(address(vow)).sig("Sin()").checked_write(3_000_000 * RAD);
+        stdstore.target(address(vow)).sig("Ash()").checked_write(2_000_000 * RAD);
+
+        address old_flap = chainLog.getAddress("MCD_FLAP");
+
+        assertEq(vow.flapper(), old_flap);
+        assertEq(vat.can(address(vow), old_flap),      1);
+        assertEq(vat.can(address(vow), address(flap)), 0);
+
+        _vote(address(spell));
+        _scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        assertEq(vow.flapper(), address(flap));
+        assertEq(vat.can(address(vow), old_flap),      0);
+        assertEq(vat.can(address(vow), address(flap)), 1);
+
+        address pip = flap.pip();
+        assertEq(pip, addr.addr("PIP_MKR"));
+
+        address pair = flap.pair();
+
+        GemAbstract mkr = GemAbstract(addr.addr("MCD_GOV"));
+
+        // Set liquidity in the pool
+        vm.prank(pauseProxy);
+        MedianAbstract(pip).kiss(address(this));
+
+        uint256 price = MedianAbstract(pip).read();
+        uint256 daiAmt = 1_000_000 * WAD;
+        GodMode.setBalance(address(dai), address(pair), daiAmt);
+        uint256 mkrAmt = 1_000_000 * WAD * WAD / price;
+        GodMode.setBalance(address(mkr), address(pair), mkrAmt * 97 / 100); // 3% worse price (should fail)
+        vm.expectRevert("FlapperUniV2/insufficient-buy-amount");
+        vow.flap();
+        GodMode.setBalance(address(mkr), address(pair), mkrAmt * 99 / 100); // Leaves just 1% worse price
+        //
+
+        uint256 initialLp = GemAbstract(pair).balanceOf(pauseProxy);
+        uint256 initialDaiVow = vat.dai(address(vow));
+        uint256 initialReserveDai = dai.balanceOf(pair);
+        uint256 initialReserveMkr = mkr.balanceOf(pair);
+
+        vow.flap();
+
+        assertGt(GemAbstract(pair).balanceOf(pauseProxy), initialLp);
```

Comment:
Why isn't this check part of the flapper itself? To my understanding, the amount of LP tokens got in return is not validated anywhere at runtime, so it can potentially be 0 and the [transaction will not revert](https://github.com/makerdao/dss-flappers/blob/6a8d6bedec27bc871a1cf1caad59014b3c0e2937/src/FlapperUniV2.sol#L198-L202)

Reply:
It's a sanity check that to be honest is not very valuable, liquidity can not be 0 due to a require in the uniswap v2 `mint` function. Anyway it doesn't harm to leave it here.